src/main/java/org/assertj/core/api/AbstractPathAssert.java

```diff
@@ -1845,6 +1900,27 @@ public SELF isNotEmptyFile() {
     return myself;
   }
 
+  /**
+   * Returns ByteArray assertions on the content of the actual {@code Path} read.
+   * <p>
+   * Example:
+   * <pre><code class='java'> Path xFile = Files.write(Paths.get("xfile.txt"), "The Truth Is Out There".getBytes());
+   *
+   * // assertion succeeds
+   * assertThat(xFile).binaryContent().isEqualTo("The Truth Is Out There".getBytes());
+   *
+   * // assertion fails:
+   * assertThat(xFile).binaryContent().isEqualTo("Elsewhere".getBytes());</code></pre>
+   *
+   * @return a ByteArrayAssert object with the binary content of the file.
+   * @throws AssertionError if the actual {@code Path} is not readable as per {@link Files#isReadable(Path)}.
+   * @throws UncheckedIOException when failing to read the actual {@code Path}.
+   */
+  public AbstractByteArrayAssert<?> binaryContent() {
+    paths.assertIsReadable(info, actual);
+    return new ByteArrayAssert(readPath()).withAssertionState(myself);
```

Comment:
similar comments as for AbstractFileAssert about sot assertions and assumptions

Reply:
Should be fixed in Should be fixed in https://github.com/assertj/assertj-core/pull/2558/commits/ca26945da0ab3ff57d5f1c0de48045b425b0156e